### DISCUSSION TITLE
Minify css and js

### DISCUSSION THREAD

**Discussion by athioak7** [sphinx-doc/sphinx#12000] [Feature requests and ideas]
It would be great if the css and js files were minified in the build or if there is the option to minify them, so that the end result is faster. My sphinx build is really big so any performance enhancements will help.

**Comment by mgeier** [comment]
Can you provide some speed measurements?
For example with your browser's developer tools or maybe with something like https://pagespeed.web.dev/?

It would be interesting where those big CSS and JS files come from. I might be core Sphinx, or your Sphinx theme, or maybe some Sphinx extension you are using?

**Comment by athioak7** [comment]
My sphinx build is local, so I can't test it, but I have some long pages that take 3-4 seconds to load. I don't necessarily have an example of a large css/js file but I thought minifying could help with the performance. I cannot manually minidy the files so I'm asking if it is possible to minify them during the build

**Reply by mgeier** [reply]
> My sphinx build is local, so I can't test it,

You can and should still test it.
Please open the developer tools of your browser and play around with them.
If you don't find what you need, try another browser.

> but I have some long pages that take 3-4 seconds to load.

Wow, that's a lot.
Sphinx's CSS and JS are probably the least of your problems.
You should analyze and see what's actually taking most of the time, and then we can start talking.

> I don't necessarily have an example of a large css/js file

Well if you don't have large files, minifying doesn't really help that much.

> but I thought minifying could help with the performance.

AFAIK, the main benefit of minifying is to reduce the size, therefore reducing the download times.
If you are talking about local files, minification won't give you much (maybe a slight improvement in parsing time).

> I cannot manually minidy the files

Well you can do that (there are tools available, just search for it) for an example page, and then you can measure if it helps or not.

And if you find a significant improvement with the CSS/JS files provided by Sphinx, then that's a great reason to consider implementing this in Sphinx itself.

> so I'm asking if it is possible to minify them during the build

This kind of performance improvements should only be done when there are measurements showing that it actually has the claimed effects.

Doing it without justification is not a good idea.
And even if it improves performance (which I'm not sure it will), we still have to consider the complexity of implementation and maintenance, and the added complexity of the installation and build process for Sphinx users.

**Reply by picnixz** [reply]
Minification of the js files should not be part of Sphinx. This can easily be achieved by a post-processing step, otherwise, we would likely have additional dependencies in the project (which we should avoid as much as possible). It would be much more easier to directly minify the produced JS file using https://www.npmjs.com/package/uglify-js and https://www.npmjs.com/package/clean-css.

As such, I'll close the discussion as "not planned".
